clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/bitbucket/BitbucketArtifactAccount.java

```diff
@@ -29,20 +30,29 @@
 
 @NonnullByDefault
 @Value
-public class BitbucketArtifactAccount implements ArtifactAccount, BasicAuth {
-  private final String name;
-  private final Optional<String> username;
-  private final Optional<String> password;
-  private final Optional<String> usernamePasswordFile;
+public class BitbucketArtifactAccount implements ArtifactAccount, BasicAuth, TokenAuth {
+  String name;
+  Optional<String> username;
+  Optional<String> password;
+  Optional<String> usernamePasswordFile;
+  Optional<String> token;
+  Optional<String> tokenFile;
 
   @Builder
   @ConstructorBinding
   @ParametersAreNullableByDefault
   BitbucketArtifactAccount(
-      String name, String username, String password, String usernamePasswordFile) {
+      String name,
+      String username,
+      String password,
+      String usernamePasswordFile,
+      String token,
```

Comment:
A minor concern - we're changing the initialization to include parameters.  Anyone extending this would get broken method signatures.  I'd use constructor overloading to reduce any risk vs. completely changing constructor signatures... (may not really be an issue, just reduces risk)

Reply:
These are configuration properties of the possible values that can be set, new ones being token and tokenFile, I think these kind of clases shouldn't be extended, in case there is an example of it being extended, I could take a look to it.
In test code it uses the Builder pattern to build the objects

Reply:
Overloaded the constructors